src/panels/lovelace/common/directives/action-handler-directive.ts

```diff
@@ -1,21 +1,31 @@
 import { directive, PropertyPart } from "lit-html";
 import "@material/mwc-ripple";
-import { LongPressOptions } from "../../../../data/lovelace";
+import {
+  ActionHandlerOptions,
+  ActionHandlerEvent,
+} from "../../../../data/lovelace";
+import { fireEvent } from "../../../../common/dom/fire_event";
 
 const isTouch =
   "ontouchstart" in window ||
   navigator.maxTouchPoints > 0 ||
   navigator.msMaxTouchPoints > 0;
 
-interface LongPress extends HTMLElement {
+interface ActionHandler extends HTMLElement {
   holdTime: number;
   bind(element: Element, options): void;
 }
-interface LongPressElement extends Element {
-  longPress?: boolean;
+interface ActionHandlerElement extends Element {
+  actionHandler?: boolean;
 }
 
-class LongPress extends HTMLElement implements LongPress {
+declare global {
+  interface HASSDomEvents {
+    action: ActionHandlerEvent;
```

Comment:
`ActionHandlerEvent` is also `{action:  string}`. So you have `{action: {action: string}}